src/shared/src/validate.rs

```diff
@@ -42,14 +42,15 @@ pub(crate) use validate_on_deserialize;
 /// - Run `test_validate_on_deserialize!($type)` to test the validation.
 #[cfg(test)]
 macro_rules! test_validate_on_deserialize {
```

Comment:
I have tweaked this macro to take the test vectors from an argument instead of a function, as otherwise I would need a separate module for every test.